requests/utils.py

```diff
@@ -548,7 +550,8 @@ def should_bypass_proxies(url):
     # exceptions we've seen, though: this call failing in other ways can reveal
     # legitimate problems.
     try:
-        bypass = proxy_bypass(netloc)
+        invalid_indicator = int(time.time() / 60)
+        bypass = _cached_proxy_bypass(netloc, invalid_indicator)
```

Comment:
So I think with this approach we *definitely* need to comment what we're doing here, to make sure that people coming down the line later understand what's going on.

I'd also love it if we tested this, though I'm not immediately sure how best to do that.